pkg/trace/stats/concentrator_test.go

```diff
@@ -410,13 +489,161 @@ func TestConcentratorSublayersStatsCounts(t *testing.T) {
 		"query|hits|env:none,resource:resource4,service:A3":                                               1,
 		"query|hits|env:none,resource:resource6,service:A3":                                               1,
 	}
+	countValsEq(t, expectedCountValByKey, receivedCounts)
+}
+
+// setupConcentrator initializes a concentrator from a given trace input.
+// calcSublayers is an option to enable sublayer metric calculation.
+func setupConcentrator(trace pb.Trace, calcSublayers bool) *Concentrator {
```

Comment:
Extracted this logic to simplify the test code.

Reply:
All these PR comments are very distracting :) The code should be self explanatory, and if something isn't clear - the reviewer will ask. I think most of us prefer a clean page of code to read through. No need to explain the motivations behind the decisions.